modules/setting/database.go

```diff
@@ -135,15 +136,19 @@ func DBConnStr() (string, error) {
 // parsePostgreSQLHostPort parses given input in various forms defined in
 // https://www.postgresql.org/docs/current/static/libpq-connect.html#LIBPQ-CONNSTRING
 // and returns proper host and port number.
-func parsePostgreSQLHostPort(info string) (string, string) {
-	host, port := "127.0.0.1", "5432"
-	if strings.Contains(info, ":") && !strings.HasSuffix(info, "]") {
-		idx := strings.LastIndex(info, ":")
-		host = info[:idx]
-		port = info[idx+1:]
-	} else if len(info) > 0 {
-		host = info
+func parsePostgreSQLHostPort(info string) (host, port string) {
+	host = info
+	h, p, err := net.SplitHostPort(info)
+	if err == nil {
+		host, port = h, p
+	}
+
+	// if it's an IPv6 address, remove the wrapper
+	if strings.HasPrefix(host, "[") && strings.HasSuffix(host, "]") {
+		host = host[1 : len(host)-1]
```

Comment:
This code seems unnecessary. `SplitHostPort` already handles the IPv6 address correctly.

Reply:
The issue is when no port is specified. Output 2 is the issue here: https://go.dev/play/p/X89w5wS6D5e

Reply:
I think a `else` is better for this case, and I managed to make some changes, let's see if the new code work well. If no, let's revert to the original code.

Reply:
From looking at it, I think it should behave the same